launch/test/launch/test_logging.py

```diff
@@ -0,0 +1,237 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the launch.logging module."""
+
+import os
+import re
+
+import launch.logging
+
+import pytest
+
+
+@pytest.fixture
+def log_dir(tmpdir_factory):
+    """Test fixture that generates a temporary directory for log files."""
+    return str(tmpdir_factory.mktemp('logs'))
+
+
+def test_bad_logging_launchConfig():
+    """Tests that setup throws at bad configuration."""
+    with pytest.raises(ValueError):
+        launch.logging.launchConfig(log_dir='not/a/real/dir')
+
+    with pytest.raises(ValueError):
+        launch.logging.launchConfig(screen_format='default', screen_style='%')
+
+    with pytest.raises(ValueError):
+        launch.logging.launchConfig(log_format='default', log_style='%')
+
+
+def test_output_loggers_bad_configuration(log_dir):
+    """Tests that output loggers setup throws at bad configuration."""
+    launch.logging.launchConfig(log_dir=log_dir)
+
+    with pytest.raises(ValueError):
+        launch.logging.getOutputLoggers('some-proc', 'not_an_alias')
+
+    with pytest.raises(ValueError):
+        launch.logging.getOutputLoggers('some-proc', {'garbage': {'log'}})
+
+    with pytest.raises(ValueError):
+        launch.logging.getOutputLoggers('some-proc', {'stdout': {'garbage'}})
+
+
+@pytest.mark.skipif(
+    pytest.config.getoption('-s') != 'no',
+    reason='seems like "-s" option is required to capture stdout from logger'
```

Comment:
Did you try to use `capsys` (I assume so since you used `capfd`)? From this doc it seems like it would work or that you could use `capsys` or `sys.stdout` conditionally by checking `capsys.disabled()`:

https://docs.pytest.org/en/latest/capture.html#accessing-captured-output-from-a-test-function

It would be nice to not skip these when `-s` is used, because I think that's the default behavior of our buildfarm, IIRC.

Reply:
> Did you try to use `capsys` (I assume so since you used `capfd`)?

Not really, I just stuck with what @jacobperron left there. Just tried `capsys` and it works like a charm. Thanks!